src/video/SDL_blit_A.c

```diff
@@ -1327,7 +1237,10 @@ static void BlitNtoNPixelAlpha(SDL_BlitInfo *info)
         DISEMBLE_RGBA(src, srcbpp, srcfmt, Pixel, sR, sG, sB, sA);
         if (sA) {
             DISEMBLE_RGBA(dst, dstbpp, dstfmt, Pixel, dR, dG, dB, dA);
-            ALPHA_BLEND_RGBA(sR, sG, sB, sA, dR, dG, dB, dA);
```

Comment:
Should we review all the places ALPHA_BLEND_RGBA() is being used, and/or change that macro directly?

Reply:
That is a good idea. I will replace the blend macro with the correct blender and remove the function call as you noted in your other comment.

I've determined all places it's used it makes sense to do, as the other nearby macros handle transforming in and out of non ARGB8888 formats. This should fix the correctness of blitting in all software rendering that involves alpha blending in the assembly, not just BlitNtoN, but I will review thoroughly to verify that claim. I'm pretty sure no current `main` code right now is using the correct whitepaper algorithm.

Reply:
Great. Can you also confirm that the algorithm used is for non-premultiplied alpha? We have a separate discussion about adding support for premultiplied alpha for a performance boost, but that hasn't been done yet.

Reply:
Can confirm it is. Alpha is assumed non-premultiplied and we perform the required fix-up step to get data in the right form. Would be interested to work on that enhancement after this PR.

Reply:
That sounds great. :)

Reply:
I've reviewed and updated all use of the macro, as well as related functions that *don't* use it, and updated a function I noticed had incorrect math to use accurate math. I also removed two blitting functions that had comments pointing out known errors in the math and replaced them with fallback calls to `BlitNtoNPixelAlpha`. Functions using the modified `ALPHA_BLEND_RGBA` macro are using them correctly in place of the classic algorithm I have proven deviates from correct.

Updating this macro took me to other files beyond `SDL_blit_A` and I reviewed their blitting algorithms. I didn't see any operations that would lead to error within `blit_0` or `blit_1`.

Reviewing `SDL_blit_auto` was difficult. I know for a fact it has errors but I'm not sure I can I fix them right now. I'll need to understand how it's generated and start from there, and I don't know a lick of perl. I know from the whitepaper that there is at least one half-bit of precision lost within blit_auto in some applications of alpha blending, unless that has been updated since 2022.

I can't make the claim all software rendering is accurate, but I can claim we improved or reached perfect accuracy in common uses of the blitting API. Indeed, no previous code I saw was using the whitepaper algorithm specifically, but many uses of math (including within `blit_auto`) appear accurate and don't introduce rounding error AFAICT.

It appears that the algorithms with error each had error from a few different sources and we are approaching consistency with accurate math.

Reply:
We can certainly replace blocks like this:
```
                ${d}R = ${s}R + ((255 - ${s}A) * ${d}R) / 255;
                ${d}G = ${s}G + ((255 - ${s}A) * ${d}G) / 255;
                ${d}B = ${s}B + ((255 - ${s}A) * ${d}B) / 255;
```
with the application of your macro.

Reply:
Your comment led me to understand I haven't fully fixed the issue outside of `blit_auto`, especially after re-reviewing the whitepaper.

The perl seems trivial to modify after your comment and working with it more. I will need to re-review everything I touched as well.

Reply:
Implemented fast multi-div in SDL_blit_auto, but it needs test coverage. Mangling the code doesn't affect any tests, leading me to think we need coverage here as well.